packages/thirdweb/src/react/core/hooks/connection/useAutoConnect.tsx

```diff
@@ -192,7 +205,16 @@ export function AutoConnect(props: AutoConnectProps) {
             message: `AutoConnect timeout : ${timeout}ms limit exceeded.`,
           });
 
-          connect(activeWallet);
+          // connected wallet could be activeWallet or smart wallet
+          const connectedWallet = await connect(activeWallet);
+
+          if (!connectedWallet) {
+            setConnectionStatus("disconnected");
+          }
+
+          if (onConnect && connectedWallet) {
+            onConnect(connectedWallet);
+          }
```

Comment:
would be easier to read here if this was instead:

```js
if(connectedWallet){
  if(onConnect){
    onConnect(connectedWallet);
  }
} else { 
  setConnectionStatus("disconnected");
}
```

or if we can return early here that could also do the trick, negation as the "first" option just always makes things difficult to read

Reply:
done